staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -312,7 +357,122 @@ func (s *APIAggregator) RemoveAPIService(apiServiceName string) {
 	s.GenericAPIServer.Handler.NonGoRestfulMux.Unregister(proxyPath)
 	s.GenericAPIServer.Handler.NonGoRestfulMux.Unregister(proxyPath + "/")
 	delete(s.proxyHandlers, apiServiceName)
+	delete(s.apiServiceSpecs, apiServiceName)
+	delete(s.toLoadAPISpec, apiServiceName)
+	s.updateOpenAPISpec()
 
 	// TODO unregister group level discovery when there are no more versions for the group
 	// We don't need this right away because the handler properly delegates when no versions are present
 }
+
+func (_ *APIAggregator) loadOpenAPISpec(p *proxyHandler, r *http.Request) (*spec.Swagger, error) {
```

Comment:
func (c *APIServiceRegistrationController) Run(stopCh <-chan struct{})  in apiservice_controller.go seems to be a background thread set up in NewWithDelegate for exactly this sort of thing.